server/src/main/java/io/spine/server/storage/memory/InMemoryInboxStorage.java

```diff
@@ -77,38 +78,62 @@ TenantInboxRecords createSlice() {
     }
 
     @Override
-    public void write(InboxMessage message) {
+    public synchronized Optional<InboxMessage> oldestMessageToDeliver(ShardIndex index) {
+        TenantInboxRecords storage = multitenantStorage.currentSlice();
+        Optional<InboxMessage> result =
+                storage.readAll()
+                       .stream()
+                       .filter((r) -> index.equals(r.getShardIndex()) &&
+                               r.getStatus() == InboxMessageStatus.TO_DELIVER)
```

Comment:
Please extract into a method or a variable with a lambda so that the alignment could be better.